lib/queue.js

```diff
@@ -0,0 +1,59 @@
+var _ = require('lodash')
+  , queue;
+
+module.exports.init = function(brain) {
+  queue = brain.deployQueue || [];
```

Comment:
the challenge here is that we will always have to make changes to the array in place. I'm thinking we should keep a reference to the brain as well in case we need to do something that will create a new array.